nemo/collections/asr/parts/preprocessing/segment.py

```diff
@@ -219,6 +219,9 @@ def from_file(
                     f"NeMo will fallback to loading via pydub."
                 )
 
+                if hasattr(audio_file, "seek"):
```

Comment:
Doesn't seem to be correct logic cause both offset and duration are ignored

Reply:
They are not ignored. They are resolved after the `audio_file.seek(0)`.
```python
        if HAVE_PYDUB and samples is None:
            try:
                samples = Audio.from_file(audio_file)
                sample_rate = samples.frame_rate
                num_channels = samples.channels
                if offset > 0:
                    # pydub does things in milliseconds
                    seconds = offset * 1000
                    samples = samples[int(seconds) :]
                if duration > 0:
                    seconds = duration * 1000
                    samples = samples[: int(seconds)]
```

Reply:
Oh woops. Missed that. Looks good.